### ISSUE TITLE
Elasticsearch 2.3 compatibility

### ISSUE THREAD

**Issue by helbow** [khituras/elasticsearch-mapper-preanalyzed#2]
Hi Erik,

I was working with your mapper plugin with Elasticsearch 2.1. Unfortunately, when I upgraded to 2.3, it turned out that the plugin is not compatible anymore. Do you have any plans on a newer version which works with 2.3 (or are you already working on it maybe)? I already forked the project but thought I might ask first, if you are working on it yourself.

Mantas

**Comment by khituras** [comment]
Hi Mantas,

first of all, thank you very much for your interest in this plugin, you're the first one except me to use it (that I know of). This might be because I never advertised it within the ElasticSearch community.

This is largely because of this very issue, I cannot keep up with the development of ElasticSearch. Those guys are just too quick for me ;-) I am sorry to tell that I have not yet an updated version of the plugin, nor have I been working on one. This repository is fully up to date. I use ElasticSearch as one piece of my larger project and since I am the only developer, I just can't manage to keep up with all the 3rd-party projects I use.

Thus, I just didn't have time to update yet and I cannot tell when or if I will work on an update. However, if you would go on with it I would gladly cooperate and help as far as my time allows.

**Comment by helbow** [comment]
Hi Erik,

thanks for the quick reply! I actually already suspected that you don't plan on updating the plugin to newer versions (and I fully understand why, elastic is really quick :-) ). I also work with Solr and absolutely like the idea of the PreAnalyzedField. As Elasticsearch doesn't provide such functionality (at least I wouldn't know) out of the box, I think this is a great plugin!

I'd really like to make this work with version 2.3.3 and maybe also later versions of Elasticsearch. Thanks for offering a helping hand!

**Comment by khituras** [comment]
Hi Mantas,

thanks for your understanding! When we could update the plugin together, that would be great. Of course I will do what I can. I don't know what has changed between ES 2.1.1 and 2.3.3. However, the plugin will be incompatible with each version change just because the version is wrong. Everything else could be just fine.

As you might have noticed, I intentionally employed the JSON format given by the Solr PreAnalyzedField. This plugin is absolutely meant as the counterpart to that Solr field type. When I started writing it, I had checked before and not found anything comparable for ElasticSearch.

If you run into any issues, don't hesitate to open a - well - issue. I'm not in any way dogmatic about my code, if anything can be made better, I'm glad to learn about it.
